### PR TITLE
Fix variable explorer when debugging and hovering

### PR DESCRIPTION
**Pull request by vandyliu**
For #6081

**Update: Rich helped me solve it.
When you go over the local variable, it sends a "hover" "update variables" message, and then gets the internal attributes of a variable like shown below in the 2nd picture. Now we check that the "update variables" message is triggered from a "scopes" message, which is when the variables update because you ran some code, etc.
15kjpCvLZf**

I tried digging into the problem first. It seems that when you hover over a variable in the IW, a new message is sent and `onDidMessageSend` is called with `message.type === 'response' && message.command === 'variables'`, so the variables are updated with that message. However the variables in the message are actually the attributes of the variable that is being hovered over, so the variables in the variable explorer get replaced by all of those variable's attributes. 
Eg. 
Code_-_Insiders_ZUaPIERIrj

Seems like a fix would be to have it so no messages send when you hover over a variable or be able to differentiate between a variable hover and a normal variable message. 

Not sure how much code to remove though for debugger variables.

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by rchiodo** [comment]
Is your intent to fix the real problem later? It seems like you're close to figuring it out.

**Comment by vandyliu** [comment]
> Is your intent to fix the real problem later? It seems like you're close to figuring it out.

I'd love to fix the real problem now honestly, but I am pretty heavily blocked. My original intentions were to try my best to figure it out and if not, disable it (like in this PR) and as stated in #6081. Then make an issue for it to be investigated later on. Maybe we can chat tomorrow on it and see if I can get a little closer?

**Comment by rchiodo** [comment]
> > Is your intent to fix the real problem later? It seems like you're close to figuring it out.
> 
> I'd love to fix the real problem now honestly, but I am pretty heavily blocked. My original intentions were to try my best to figure it out and if not, disable it (like in this PR) as stated in #6081. Then make an issue for it to be investigated later on. Maybe we can chat tomorrow on it and see if I can get a little closer?

Sounds good. Ping me on Teams.

File: src/client/datascience/jupyter/debuggerVariables.ts [hunk 1]
```diff
@@ -207,17 +210,53 @@ export class DebuggerVariables extends DebugLocationTracker
...
+            message.type === 'response' &&
+            message.command === 'variables' &&
+            message.body &&
+            this.currentSeqNumsForVariables.has(message.request_seq)
+        ) {
             // If using the interactive debugger, update our variables.
             // eslint-disable-next-line
             // TODO: Figure out what resource to use
+
+            // Only update variables if it came from a "scopes" command and not a "hover"
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/debuggerVariables.ts:253]: Thanks for the comment here. I need to brush up on the DAP 😆

File: src/client/datascience/jupyter/debuggerVariables.ts [hunk 2]
```diff
@@ -207,17 +210,53 @@ export class DebuggerVariables extends DebugLocationTracker
...
     // This special DebugAdapterTracker function listens to messages sent from the debug adapter to VS Code
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     public onDidSendMessage(message: any) {
         super.onDidSendMessage(message);
         // When the initialize response comes back, indicate we have started.
         if (message.type === 'response' && message.command === 'initialize') {
             this.debuggingStarted = true;
-        } else if (message.type === 'response' && message.command === 'variables' && message.body) {
+        } else if (message.type === 'response' && message.command === 'scopes' && message.body && message.body.scopes) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/debuggerVariables.ts:234]: Do you need this eslint disable here?

File: src/client/datascience/jupyter/debuggerVariables.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/client/datascience/jupyter/debuggerVariables.ts:253]: Looks legit to me but might want to wait for the debugger expert signoff.

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/datascience/jupyter/debuggerVariables.ts (hunk 2 above)
Comment (vandyliu) [reply src/client/datascience/jupyter/debuggerVariables.ts]: 😅 Missed that one, ty!

**Review by rchiodo — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

**Pull request merged**